### PR TITLE
Fix docs workflow

### PR DESCRIPTION
**Pull request by ADBond**
It turns out the issue with the docs workflow was a clash between two plugins: `mknotebooks` and `include-markdown` (it appears they _may_ run together, but _very_ intensive and takes > 2 hours so not useable in practice).

This PR includes a manual pre-processing script to mimic the functionality of `include-markdown`. The included syntax is the same (without flexibility) - we simply look for the tag within docs `.md` files, and replace the tags with the content of the supplied path. It is not hugely robust, but should be fine given our currently minimal use of inclusions.

It is primarily designed to be run in CI - running manually alters the files and so runs the _slight_ risk that altered files get committed to the repo (and thus become insensitive to future code changes), but this should be picked up in review, and I imagine the script will be used pretty minimally anyhow. At any rate if it becomes annoying (and the issue with the plugins remains unresolved) then we could always wrap the functionality of the script in [our own minimal mkdocs plugin](https://www.mkdocs.org/dev-guide/plugins/).

### CODE REVIEW

**Comment by ADBond** [comment]
[Workflow run](https://github.com/ADBond/splink/actions/runs/4260875809) and [results thereeof](https://github.com/ADBond/splink/tree/gh-pages).

File: scripts/make_docs_locally.sh [hunk 1]
```diff
@@ -4,16 +4,16 @@
...
-# fi
+UPDATE="TRUE"
+if [[ $UPDATE == "TRUE" ]]
+then
+    rm -rf docs/demos/
+    git clone https://github.com/moj-analytical-services/splink_demos.git docs/demos/
+fi
 
 deactivate
 python3 -m venv docs-venv
```
Comment (ThomasHepworth) [inline scripts/make_docs_locally.sh:19]: Do we want to add `python3 scripts/preprocess_markdown_includes.py` to this script?

File: scripts/preprocess_markdown_includes.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import re
```
Comment (ThomasHepworth) [inline scripts/preprocess_markdown_includes.py:1]: Include a note that if you're running/checking this manually, you need to ensure you're in the root folder of your project.

File: scripts/make_docs_locally.sh (hunk 1 above)
Comment (ADBond) [reply scripts/make_docs_locally.sh:19]: This is a good point - oversight on my part.

File: scripts/make_docs_locally.sh [hunk 3]
```diff
@@ -4,22 +4,22 @@
...
 source docs-venv/bin/activate
 pip install --upgrade pip
 pip install -r scripts/docs-requirements.txt
 
+# manually preprocess include files as include-markdown plugin clashes with mknotebooks
+# make sure not to commit changes to files with inclusions!
+python3 scripts/preprocess_markdown_includes.py
+
```
Comment (ThomasHepworth) [inline scripts/make_docs_locally.sh:23]: You need to ensure you've run `python3 scrips/generate_dialect_comparison_docs.py` prior to this step

File: scripts/make_docs_locally.sh [hunk 4]
```diff
@@ -4,22 +4,22 @@
...
+then
+    rm -rf docs/demos/
+    git clone https://github.com/moj-analytical-services/splink_demos.git docs/demos/
+fi
 
 deactivate
 python3 -m venv docs-venv
 source docs-venv/bin/activate
 pip install --upgrade pip
 pip install -r scripts/docs-requirements.txt
```
Comment (ThomasHepworth) [inline scripts/make_docs_locally.sh:19]: I'm happy not to include this, but it ensures `generate_dialect_comparison_docs.py` gets run as expected.

```suggestion
if [[ ! -f "docs/includes/generated_files/comparison_level_library_dialect_table.md" ]]
then
    pip install poetry
    poetry install
    python3 scripts/generate_dialect_comparison_docs.py
fi
```

**Review by ThomasHepworth — APPROVED** [review]
See my final comment. Include it if you feel it's useful.

Otherwise, feel free to merge!

**Pull request merged**